Fontify based on rendered column widths

bcc32 opened the pull request:
When tabulated-list-mode renders a line, it considers the remaining
available space in the selected window and may reduce the width of a
column compared to the width specified in tabulated-list-format.  As a
result, fontifying or otherwise calculating offsets based on the
widths in tabulated-list-format can lead to colors bleeding over into
the next column.

Delete function pocket-reader--column-data, which returns the
"theoretical" start and end positions of each column, and instead use
the actual text properties in the buffer.

Fixes #36.

bcc32 committed:
Fontify based on rendered column widths

When tabulated-list-mode renders a line, it considers the remaining
available space in the selected window and may reduce the width of a
column compared to the width specified in tabulated-list-format.  As a
result, fontifying or otherwise calculating offsets based on the
widths in tabulated-list-format can lead to colors bleeding over into
the next column.

Delete function pocket-reader--column-data, which returns the
"theoretical" start and end positions of each column, and instead use
the actual text properties in the buffer.

Fixes #36.

alphapapa commented:
Thanks.

I don't see the `text-property-search` library in some Emacs versions I'm using, as well as the `prop-match-beginning/end` functions.  What version were they added in?  If we use them, we'll have to increase the minimum Emacs version.

bcc32 commented:
Ah, good catch.  They were introduced in emacs 27.1.  I can also reimplement them using `next-single-property-change`, which is much older (I'm not sure when it was added, but they got a new function argument 19.22, so it's safe to say they're very old).

bcc32 committed:
Remove dependency on text-property-search

text-property-search was added in Emacs 27.1, so just reimplement the
functionality needed here.

alphapapa committed:
Docs: Update changelog

alphapapa commented:
Thanks.
